pyiron_contrib/atomistics/atomistics/job/trainingcontainer.py

```diff
@@ -172,9 +174,26 @@ def run_if_interactive(self):
 
     def to_hdf(self, hdf=None, group_name=None):
         super().to_hdf(hdf=hdf, group_name=group_name)
-        with catch_warnings():
-            self._table.to_hdf(self.project_hdf5.file_name, self.name + "/output/structure_table")
+        self._container.to_hdf(self.project_hdf5, "structures")
 
     def from_hdf(self, hdf=None, group_name=None):
         super().from_hdf(hdf=hdf, group_name=group_name)
-        self._table = pd.read_hdf(self.project_hdf5.file_name, self.name + "/output/structure_table")
+        hdf_version = self.project_hdf5.get("HDF_VERSION", "0.1.0")
+        if hdf_version == "0.1.0":
+            table = pd.read_hdf(self.project_hdf5.file_name, self.name + "/output/structure_table")
+            self.include_dataset(table)
+        else:
+            self._container = StructureContainer()
```

Comment:
I think this line is not needed? `__init__` should take care of it before `from_hdf` gets called.

Reply:
This is forces `_container` to be empty again, if you ever call `from_hdf` on a `TrainingContainer`.  It's an edge case for sure, though.